### ISSUE TITLE
How is `useSuspenseQuery` supposed to work with Next App Router `Loading` files?

### ISSUE THREAD

**Issue by rob-strover-axomic** [apollographql/apollo-client-nextjs#142]
Hello,

Thanks for all your awesome work.

I'm using the Next JS 13 app router with the `@apollo/experimental-nextjs-app-support/ssr` `useSuspenseQuery` hook in one of my client component `page.tsx` files. When I load the page, I can see in the network tab in chrome that the server is responding with the `loading.tsx` content and doesn't seem to wait for the query to complete and render the page content before responding. Is this expected? If I remove the `loading.tsx` file then the server does wait for the query to complete and responds with the rendered `page.tsx` content. Ideally I do want a loading state to appear on the client side as a placeholder when navigating on the client side.

To summarise; Ideally, I want the server to wait for the query to complete, render the content and then respond. I would then want client side route changes to show the loading content whilst the frontend waits for the request to complete. Is this possible?

Thanks for taking the time to give thoughts here, is there something I'm missing?

Code samples below:
```typescript
// ApolloProvider.tsx
"use client";

import React from 'react';

import { ApolloLink, HttpLink } from '@apollo/client';
import {
    ApolloNextAppProvider,
    NextSSRInMemoryCache,
    NextSSRApolloClient,
    SSRMultipartLink,
} from '@apollo/experimental-nextjs-app-support/ssr';

const makeClient = () => {
    const httpLink = new HttpLink({
        uri: '----',
        headers: { 'Access-Control-Allow-Origin': '*' },
        fetchOptions: { cache: "no-store" },
    });

    return new NextSSRApolloClient({
        cache: new NextSSRInMemoryCache(),
        link: typeof window === "undefined"
            ? ApolloLink.from([
                new SSRMultipartLink({
                    stripDefer: true,
                }),
                httpLink,
            ])
            : httpLink
    });
}

interface Props {
    children?: React.ReactNode;
}

const ApolloProvider = ({ children }: Props) => (
    <ApolloNextAppProvider makeClient={makeClient}>
        {children}
    </ApolloNextAppProvider>
);

export default ApolloProvider;

```

```typescript
// root layout.tsx
import React from 'react';

import ApolloProvider from 'path/to/ApolloProvider';

interface RootLayoutProps {
    children: React.ReactNode;
    params: { lang: Locale };
}

export default async function RootLayout({
    children,
}: RootLayoutProps) {
    return (
        <html>
            <body>
                <ApolloProvider>
                    {children}
                </ApolloProvider>
            </body>
        </html>
    );
}

```

```typescript
// page.tsx
'use client';

import React from 'react';

import { gql } from '@apollo/client';
import { useSuspenseQuery } from '@apollo/experimental-nextjs-app-support/ssr';

const PAGE_QUERY = gql`
    query GetData($id: ID!) {
        ...
    }
`;

export const dynamic = "force-dynamic";

export default function Page({ params }: { params: { id: string } }) {
    const { id } = params;
    
    const { error, data } = useSuspenseQuery<GetPageQuery, GetPageQueryVariables>(PAGE_QUERY, {
        variables: { id },
    })

    if (error || !data) {
          // failure ui here
    }

   return (
        <div>
            {data.foo.bars.map(bar => (....)}
       </div>
   );
```

```typescript
// loading.tsx

"use client"

import React from 'react';

const Loading = () => {
    return <p>Loading...</p>
}

export default Loading
```

**Comment by rob-strover-axomic** [comment]
@phryneas Thanks for your speedy response!

So am I right in saying there is no way to produce the use case I described above? Does this mean it is not possible to retrieve data in an SEO friendly way outside of serverside components?

**Comment by phryneas** [comment]
You will get the same result with Server Components once you have a few Suspense components in there.

This is React's "streaming render".

I think I read somewhere that Next detects if it's accessed by a crawler and doesn't do out-of-order streaming in that case, but I might be wrong.

**Comment by phryneas** [comment]
This discussion might be interesting to you: https://github.com/vercel/next.js/discussions/50829

**Comment by rob-strover-axomic** [comment]
@phryneas Thanks, this is helpful, strange how lighthouse seems to get a fully server side rendered version of the site back. I will try and simulate what's going on here with Postman.

I'm still confused as to how streaming pieces of UI rendered on the server is better than using the client to render content after the first load but maybe that answer will become clear as I keep going.

**Comment by rob-strover-axomic** [comment]
Thanks for all your help and information here. :)

**Comment by Gytjarek** [comment]
@rob-strover-axomic were you able to implement it? Could you tell how you done it in your app?

**Comment by rob-strover-axomic** [comment]
Hi @Gytjarek 

Yes, I implemented according to the [usage instructions in the readme](https://github.com/apollographql/apollo-client-nextjs?tab=readme-ov-file#usage). The problems I was having with page responsiveness were actually down to a slow backend service that I needed to call on each route change in order to get the metadata content for my page components (server components). Having subsequent route changes after initial page load would require substantial reworking and would mean not leveraging the NextJS App router as intended.

One thing I would recommend you do is move to Next 14. This seemed to help with some unexpected routing behaviour I was coming up against.

Hope this helps
